crates/runtime/src/http/v1/chat.rs

```diff
@@ -50,51 +51,62 @@ pub(crate) async fn post(
     )
     .label("model".to_string(), req.model.clone());
 
-    let model_id = req.model.clone();
-    match llms.read().await.get(&model_id) {
-        Some(model) => {
-            if req.stream.unwrap_or_default() {
-                match model.write().await.chat_stream(req).await {
-                    Ok(strm) => {
-                        create_sse_response(strm, time::Duration::from_secs(30), Some(span))
-                    }
-                    Err(e) => {
-                        span.with_error_message(e.to_string()).finish();
-                        tracing::debug!("Error from v1/chat: {e}");
-                        StatusCode::INTERNAL_SERVER_ERROR.into_response()
-                    }
-                }
-            } else {
-                match model.write().await.chat_request(req).await {
-                    Ok(response) => {
-                        let preview = response
-                            .choices
-                            .first()
-                            .map(|s| serde_json::to_string(s).unwrap_or_default())
-                            .unwrap_or_default();
-                        span.truncated_output_text(Arc::from(preview)).finish();
-                        Json(response).into_response()
+    let tspan = tracing::span!(target: "task_history", tracing::Level::INFO, "ai_completion", input_text = %serde_json::to_string(&req).unwrap_or_default());
```

Comment:
I don't know if it's possible, but it would be convenient if the `"ai_completion"` span was produced within the `Box<dyn Chat>`'s `.chat_request` & `.chat_stream` functions. We call this method in many places (e.g. recursively in `ToolUsingChat`, and right now we'd have to keep track of all of them, and keep parameters in sync.